packages/edit-navigation/src/store/test/resolvers.js

```diff
@@ -28,288 +37,163 @@ jest.mock( '@wordpress/blocks', () => {
-				{
-					attributes: {
-						orientation: 'vertical',
-					},
-					clientId: expect.stringMatching( /client-id-\d+/ ),
-					innerBlocks: [
-						{
-							attributes: {
-								__internalRecordId: 100,
-								label: 'wp.com',
-								url: 'http://wp.com',
-								className: 'menu classes',
-								rel: 'nofollow',
-								description: 'description',
-								title: 'link title',
-								kind: 'custom',
-							},
-							clientId: 'client-id-0',
-							innerBlocks: [],
-							name: 'core/navigation-link',
-						},
-						{
-							attributes: {
-								__internalRecordId: 101,
-								label: 'wp.org',
-								url: 'http://wp.org',
-								opensInNewTab: true,
-								kind: 'custom',
-							},
-							clientId: 'client-id-1',
-							innerBlocks: [],
-							name: 'core/navigation-link',
-						},
-						{
-							attributes: {
-								__internalRecordId: 102,
-								label: 'My Example Page',
-								url: '/my-example-page/',
-								opensInNewTab: true,
-								kind: 'post-type',
-								type: 'page',
-								id: 56789,
-							},
-							clientId: 'client-id-2',
-							innerBlocks: [],
-							name: 'core/navigation-link',
-						},
-					],
-					name: 'core/navigation',
-				},
-			],
-			meta: {
-				menuId,
-			},
-		};
+	it( 'returns early when a menuId is not provided', async () => {
+		const menuId = null;
 
-		expect( generator.next( menuItems ).value ).toEqual(
-			dispatch(
-				'core',
-				'receiveEntityRecords',
-				NAVIGATION_POST_KIND,
-				NAVIGATION_POST_POST_TYPE,
-				navigationBlockStubPost,
-				{ id: navigationBlockStubPost.id },
-				false
-			)
-		);
+		const registry = createRegistryWithStores();
+		await registry
+			.resolveSelect( editNavigationStore )
+			.getNavigationPostForMenu( menuId );
 
-		expect( generator.next().value ).toEqual(
-			dispatch( 'core', 'finishResolution', 'getEntityRecord', [
-				NAVIGATION_POST_KIND,
-				NAVIGATION_POST_POST_TYPE,
-				stubPost.id,
-			] )
-		);
+		const stubPost = registry
+			.select( coreStore )
+			.getEntityRecord( 'root', 'postType', menuId );
 
-		expect( generator.next().done ).toBe( true );
+		expect( stubPost ).toBeFalsy();
 	} );
 
-	it( 'creates correct core/navigation-link block variations from menu objects', () => {
+	it( 'creates a stub navigation post for menu id', async () => {
 		const menuId = 123;
 
-		const generator = getNavigationPostForMenu( menuId );
+		const registry = createRegistryWithStores();
+		const stubPost = await registry
+			.resolveSelect( editNavigationStore )
+			.getNavigationPostForMenu( menuId );
 
-		// Advance generator
-		generator.next(); // Gen step: yield persistPost
-		generator.next(); // Gen step: yield dispatch "getEntityRecord"
-		generator.next(); // Gen step: yield resolveMenuItems
+		expect( stubPost ).toBeTruthy();
+		expect( stubPost.id ).toBe( 'navigation-post-123' );
+		expect( stubPost.blocks ).toHaveLength( 1 );
+		expect( stubPost.blocks[ 0 ].innerBlocks ).toHaveLength( 3 );
```

Comment:
We're not really asserting a lot here. Maybe `expect( stubPost ).toMatchSnapshot()`? I overuse snapshots because they give you a lot for zero effort 😂